common/beerocks/bcl/source/beerocks_socket_thread.cpp

```diff
@@ -261,6 +274,11 @@ bool socket_thread::work()
         return false;
     }
 
+    // If select was skipped, rest the select timeout to default value
+    if (m_skip_next_select_timeout) {
+        set_select_timeout(m_select_timeout_msec);
+    }
+
```

Comment:
This is a bit confusing. At least update the flag here:
```suggestion
    // If select was skipped, rest the select timeout to default value
    if (m_skip_next_select_timeout) {
        m_skip_next_select_timeout = false;
        set_select_timeout(m_select_timeout_msec);
    }
```